### PR TITLE
Fixes for release 0.6.0 -> release 0.7.0

### PR DESCRIPTION
**Pull request by nonik0**
## Description
Fixes for #356.
- Update to .net8 to resolve nuget dependency issue with Azure.Identity and CG requirement
- Update to latest Bicep version
- Update to PSRule and related test updates and new test
- Fixes for nuspec files, nuget files packed and sanity checked

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [x] The pull request does not introduce breaking changes.

### [General Guidelines](../CONTRIBUTING.md)
- [x] Title of the pull request is clear and informative.
- [x] Description of the pull request is clear and informative.
- [x] I have added myself to the 'assignees'.
- [x] I have added 'linked issues' if relevant.

### [Testing Guidelines](../CONTRIBUTING.md#tests)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/Analyzer.Cli.NuGet/Azure.Templates.Analyzer.CommandLine.win-x64.0.6.1.nupkg.zip
Comment (JohnathonMohr) [inline src/Analyzer.Cli.NuGet/Azure.Templates.Analyzer.CommandLine.win-x64.0.6.1.nupkg.zip:1]: Remove

File: src/Analyzer.PowerShellRuleEngine/PowerShellRuleEngine.cs [hunk 1]
```diff
@@ -115,13 +118,16 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(TemplateContext templateContext)
                     },
                     Execution = new ExecutionOption
                     {
-                        NotProcessedWarning = false,
```
Comment (VeraBE) [inline src/Analyzer.PowerShellRuleEngine/PowerShellRuleEngine.cs:118]: Was this removed intentionally? I don't fully remember why it was set but I think there was a reason

File: src/Analyzer.PowerShellRuleEngine/PowerShellRuleEngine.cs [hunk 2]
```diff
@@ -115,13 +118,16 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(TemplateContext templateContext)
...
+                        UnprocessedObject = ExecutionActionPreference.Ignore,
                         // PSRule internally creates a PowerShell initial state with InitialSessionState.CreateDefault().
                         // SessionState.Minimal causes PSRule to use CreateDefault2 instead of CreateDefault.
-                        InitialSessionState = PSRule.Configuration.SessionState.Minimal
+                        InitialSessionState = SessionState.Minimal
                     }
                 };
+
+                // placeholder value for location is westus2
+                optionsForFileAnalysis.Configuration["AZURE_RESOURCE_ALLOWED_LOCATIONS"] = new[] { "westus2" };
```
Comment (VeraBE) [inline src/Analyzer.PowerShellRuleEngine/PowerShellRuleEngine.cs:129]: Is there any reason to leave this here?

File: src/Analyzer.PowerShellRuleEngine/PSRuleHostContext.cs [hunk 3]
```diff
@@ -41,6 +41,11 @@ public PSRuleHostContext(TemplateContext templateContext, ILogger logger = null)
                 : new JsonSourceLocationResolver(templateContext);
         }
 
+        public override ActionPreference GetPreferenceVariable(string variableName)
+        {
+            return variableName == "VerbosePreference" ? ActionPreference.Continue : base.GetPreferenceVariable(variableName);
```
Comment (JohnathonMohr) [inline src/Analyzer.PowerShellRuleEngine/PSRuleHostContext.cs:46]: Does this need to be case sensitive? Suggest using String.Equals() method.

**Review by JohnathonMohr — CHANGES_REQUESTED** [review]

File: src/Analyzer.Core.UnitTests/TemplateAnalyzerTests.cs [hunk 4]
```diff
@@ -28,7 +28,7 @@ public static void AssemblyInitialize(TestContext context)
         [DataTestMethod]
         [DataRow(@"{ ""azureActiveDirectory"": { ""tenantId"": ""tenantId"" } }", "Microsoft.ServiceFabric/clusters", 1, 1, DisplayName = "1 matching resource with 1 passing evaluation")]
         [DataRow(@"{ ""azureActiveDirectory"": { ""someProperty"": ""propertyValue"" } }", "Microsoft.ServiceFabric/clusters", 1, 0, DisplayName = "1 matching resource with 1 failing evaluation")]
-        [DataRow(@"{ ""property1"": { ""someProperty"": ""propertyValue"" } }", "Microsoft.MachineLearningServices/workspaces", 0, 0, DisplayName = "0 matching resources with no results")]
+        [DataRow(@"{ ""property1"": { ""someProperty"": ""propertyValue"" } }", "Microsoft.MachineLearningServices/workspaces", 2, 0, DisplayName = "0 matching resources with no results")]
```
Comment (VeraBE) [inline src/Analyzer.Core.UnitTests/TemplateAnalyzerTests.cs:31]: Nit: the display name

File: src/Analyzer.PowerShellRuleEngine/PowerShellRuleEngine.cs (hunk 1 above)
Comment (nonik0) [reply src/Analyzer.PowerShellRuleEngine/PowerShellRuleEngine.cs:118]: No, this option was renamed/deprecated. See the attribute "UnprocessedObject = ExecutionActionPreference.Ignore".

File: src/Analyzer.PowerShellRuleEngine/PowerShellRuleEngine.cs (hunk 2 above)
Comment (nonik0) [reply src/Analyzer.PowerShellRuleEngine/PowerShellRuleEngine.cs:129]: Yes, this changes the PSRule validation behavior to expect westus2 as an expected region, where the default is eastus. Without this we would get the AZR-000167 false positive as before.

File: src/Analyzer.PowerShellRuleEngine/PSRuleHostContext.cs [hunk 5]
```diff
@@ -41,6 +41,12 @@ public PSRuleHostContext(TemplateContext templateContext, ILogger logger = null)
                 : new JsonSourceLocationResolver(templateContext);
         }
 
+        /// <inheritdoc/>
+        public override ActionPreference GetPreferenceVariable(string variableName)
+        {
+            return variableName.Equals("VerbosePreference", System.StringComparison.OrdinalIgnoreCase) ? ActionPreference.Continue : base.GetPreferenceVariable(variableName);
```
Comment (JohnathonMohr) [inline src/Analyzer.PowerShellRuleEngine/PSRuleHostContext.cs:47]: Not sure how this method is used, but consider adding null-safety here.

**Review by JohnathonMohr — APPROVED** [review]
One more small suggestion.

File: src/Analyzer.PowerShellRuleEngine/PSRuleHostContext.cs (hunk 5 above)
Comment (nonik0) [reply src/Analyzer.PowerShellRuleEngine/PSRuleHostContext.cs]: static method works for this, has null checks

Comment (nonik0) [reply src/Analyzer.PowerShellRuleEngine/PSRuleHostContext.cs]: Also, these methods enable PSRule logging with verbose option. Helped me to debug issue with PSRule with Bernie.

**Review by VeraBE — APPROVED** [review]

File: src/Analyzer.Core.NuGet/Analyzer.Core.nuspec [hunk 6]
```diff
@@ -12,16 +12,13 @@
 
     <dependencies>
       <group targetFramework="net7.0">
```
Comment (JohnathonMohr) [inline src/Analyzer.Core.NuGet/Analyzer.Core.nuspec:14]: net8.0

**Review by reynoldsa — APPROVED** [review]
Looks good now that net8 is mentioned in the notes.

**Pull request merged**